automated-comments/csharp/general/return_immediately.md

```diff
@@ -0,0 +1 @@
+Instead of storing the return value into an intermediate value, you could just return it directly. This usually makes code a bit easier to understand, as seeing that a method returns will allow you to skip reading the rest of the code, which you would still have to do when a variable is used, as that variable could be modified in another line of code.
```

Comment:
@mikedamay said:

"I have slightly milder reservations about “Instead of storing the return value into an intermediate value, you could just return it directly” than I do about the “nested if” but I don’t think I would make the point in a review. There is even a school of thought that a single return statement is a good thing although I think that’s going out of fashion. My own position is that it depends on the specific case – length of method, readability, for starters."

Reply:
Just to be sure, this comment will only be applied when the code looks like this:

```csharp
public int Test()
{
    ....
    var ret = CalculateValue();
    return ret;
}
```

So there are two conditions that have to be met:

- The value that is returned is assigned to in the previous line.
- The return statement is the last statement in the method.

Does it make more sense now to have this comment? I think I may have accidentally copied the wrong comment here, hence the confusion. To reduce the confusion, I've renamed the comment to "do_not_assign_and_return". Does that make things more clear?

Regarding the direct return that I think you are referring to, I _am_ in favor of doing that, but I agree that that can be a very personal thing.

Reply:
I totally agree with your position.

Reply:
Good to hear!